crates/state/src/account_info.rs

```diff
@@ -117,6 +117,40 @@ impl AccountInfo {
         self
     }
 
+    /// Sets the [`AccountInfo`] `balance`.
+    #[inline]
+    pub fn set_balance(&mut self, balance: U256) -> &mut Self {
+        self.balance = balance;
+        self
+    }
+
+    /// Sets the [`AccountInfo`] `nonce`.
+    #[inline]
+    pub fn set_nonce(&mut self, nonce: u64) -> &mut Self {
+        self.nonce = nonce;
+        self
+    }
+
+    /// Sets the [`AccountInfo`] `code_hash` and clears any cached bytecode.
```

Comment:
we should also add a note here, because this could be called incorrectly with `set_code().set_code_hash()`

Reply:
I feel you on that, @mattsse. updated the comment in latest commit https://github.com/bluealloy/revm/pull/2429/commits/604869effb264627defdf3e21e120fc30b4962f5